.buildkite/common.py

```diff
@@ -186,8 +186,11 @@ def ab_revision_build(revision):
         f"if ! git cat-file -t $$commitish; then commitish=origin/{revision}; fi",
         "branch_name=tmp-$$commitish",
         "git branch $$branch_name $$commitish",
+        f"if [ ! -d build/{revision} ]; then",
         f"git clone -b $$branch_name . build/{revision}",
         f"cd build/{revision} && ./tools/devtool -y build --release && cd -",
+        "git branch -D $$branch_name",
+        "fi",
```

Comment:
Hmmm, does this mean that if (for whatever reason) the branch folder already exists we will not build? If this happens, does it mean we will use an old checkout? It looks like it.

I guess this cannot really happen since we later delete it, but (unless I 'm missing something) the code like that does not reflect our intention

Reply:
Actually, mh, we can't delete the repository because the upload step needs it. I guess we can just have a simple step that checks if "revision a == revision b" and in this case only does the clone + build thing once